java-connectors/kafka-connect-common/src/main/java/io/lenses/streamreactor/common/util/JarManifest.java

```diff
@@ -44,51 +45,50 @@ public class JarManifest {
+  private Map<String, String> jarAttributes = new HashMap<>();
 
   /**
    * Creates JarManifest.
    * @param location Jar file location
    */
   public JarManifest(URL location) {
-    Manifest manifest;
-
-    try (JarFile jarFile = new JarFile(new File(location.toURI()))) {
-      manifest = jarFile.getManifest();
+    try {
+      File file = new File(location.toURI());
+      if (file.isFile()) {
+        JarFile jarFile = new JarFile(file);
+        ofNullable(jarFile.getManifest()).flatMap(mf -> of(mf.getMainAttributes()))
+            .ifPresent(mainAttrs -> jarAttributes = extractMainAttributes(mainAttrs));
+        jarFile.close();
+      }
     } catch (URISyntaxException | IOException e) {
       throw new ConnectorStartupException(e);
     }
-    extractMainAttributes(manifest.getMainAttributes());
   }
 
   /**
    * Creates JarManifest.
    * @param jarFile
    */
   public JarManifest(JarFile jarFile) {
-    Manifest manifest;
     try {
-      Optional<JarFile> jarFileOptional = of(jarFile);
-      manifest = jarFileOptional.get().getManifest();
-    } catch (NullPointerException | IOException e) {
+      Optional<JarFile> jarFileOptional = ofNullable(jarFile);
+      if (jarFileOptional.isPresent()) {
+        ofNullable(jarFileOptional.get().getManifest()).flatMap(mf -> of(mf.getMainAttributes()))
+            .ifPresent(mainAttrs -> jarAttributes = extractMainAttributes(mainAttrs));
+      }
+    } catch (IOException e) {
       throw new ConnectorStartupException(e);
     }
-    extractMainAttributes(manifest.getMainAttributes());
   }
 
-  private void extractMainAttributes(Attributes mainAttributes) {
-    jarAttributes.put(REACTOR_VER.getAttributeName(),
-        ofNullable(mainAttributes.getValue(REACTOR_VER.getAttributeName())).orElse(UNKNOWN));
-    jarAttributes.put(KAFKA_VER.getAttributeName(),
-        ofNullable(mainAttributes.getValue(KAFKA_VER.getAttributeName())).orElse(UNKNOWN));
-    jarAttributes.put(GIT_REPO.getAttributeName(),
-        ofNullable(mainAttributes.getValue(GIT_REPO.getAttributeName())).orElse(UNKNOWN));
-    jarAttributes.put(GIT_HASH.getAttributeName(),
-        ofNullable(mainAttributes.getValue(GIT_HASH.getAttributeName())).orElse(UNKNOWN));
-    jarAttributes.put(GIT_TAG.getAttributeName(),
-        ofNullable(mainAttributes.getValue(GIT_TAG.getAttributeName())).orElse(UNKNOWN));
-    jarAttributes.put(REACTOR_DOCS.getAttributeName(),
-        ofNullable(mainAttributes.getValue(REACTOR_DOCS.getAttributeName())).orElse(UNKNOWN));
+
+  private Map<String, String> extractMainAttributes(Attributes mainAttributes) {
+    final Map<String, String> attributesFromJar = new HashMap<>();
+    Arrays.stream(ManifestAttributes.values()).forEach(manifestAttribute ->
+        attributesFromJar.put(manifestAttribute.getAttributeName(),
+            ofNullable(mainAttributes.getValue(manifestAttribute.getAttributeName())).orElse(UNKNOWN))
+    );
+    return attributesFromJar;
   }
```

Comment:
🚀 

How do you feel about the below tweak to avoid the mutable map state?

```suggestion
  private Map<String, String> extractMainAttributes(Attributes mainAttributes) {
    final Map<String, String> attributesFromJar = new HashMap<>();
    Arrays.stream(ManifestAttributes.values()).forEach(manifestAttribute ->
        attributesFromJar.put(manifestAttribute.getAttributeName(),
            ofNullable(mainAttributes.getValue(manifestAttribute.getAttributeName())).orElse(UNKNOWN))
    );
    return attributesFromJar;
  }
```

Reply:
I'm indifferent to for it to be immutable to be honest: I don't see the requirement, nor I have anything against in this scenario :)
I've changed that using your suggestion